src/Core/Silk.NET.Core/Native/SilkMarshal.cs

```diff
@@ -512,19 +512,43 @@ public static string[] MemoryToStringArray
             Func<nint, string> customUnmarshaller
         ) => PtrToStringArray(input, input.Length / IntPtr.Size, customUnmarshaller);
 
-        private static unsafe string Utf8PtrToString(nint ptr)
+        /// <summary>
+        /// Gets the length of the given native string.
+        /// </summary>
+        /// <param name="ptr">The native string pointer.</param>
+        /// <param name="encoding">The encoding.</param>
+        /// <returns>The length of the string.</returns>
+#if NET6_0_OR_GREATER
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static unsafe nuint StringLength(
+            nint ptr,
+            NativeStringEncoding encoding = NativeStringEncoding.Ansi
+        ) =>
+            (nuint)(
+                encoding == NativeStringEncoding.LPWStr
+                    ? MemoryMarshal.CreateReadOnlySpanFromNullTerminated((char*)ptr).Length
+                    : MemoryMarshal.CreateReadOnlySpanFromNullTerminated((byte*)ptr).Length
+            );
+#else
+        public static unsafe nuint StringLength(
+            nint ptr, 
+            NativeStringEncoding encoding = NativeStringEncoding.Ansi
+        )
         {
-            var span = new Span<byte>((void*) ptr, int.MaxValue);
-            span = span.Slice(0, span.IndexOf(default(byte)));
-            if (span.Length == 0)
+            if (ptr == 0)
             {
-                return string.Empty;
+                return 0;
             }
+            nuint ret;
+            for (ret = 0; ((byte*) ptr)![ret] != 0; ret++) { }
```

Comment:
This would be broken for `NativeStringEncoding.LPWStr`, or is that never used < net6?

Reply:
Good point. This codepath specifically isn't hit today, but if we're claiming to support multiple encodings we should implement it properly. Resolved in latest commit.